store/sqlstore/store.go

```diff
@@ -1176,6 +1185,24 @@ func (ss *SqlStore) ensureMinimumDBVersion(ver string) (bool, error) {
 	return true, nil
 }
 
+func (ss *SqlStore) ensureDatabaseCollation() error {
+	if *ss.settings.DriverName == model.DatabaseDriverMysql {
+		var dbCollation struct {
+			Variable_name string
+			Value         string
+		}
+		if err := ss.GetMasterX().Get(&dbCollation, "SHOW VARIABLES LIKE 'collation_database'"); err != nil {
+			return errors.Wrap(err, "unable to select variables")
+		}
+
+		if _, ok := invalidMySQLCollations[dbCollation.Value]; ok {
+			mlog.Warn("Database collation is not compatible", mlog.String("expected", "utf8mb4_general_ci"), mlog.String("found", dbCollation.Value))
```

Comment:
I think it's good to remove the expected value. There can be several "right" collation values.

Also, what is the error we see while using `utf8mb4_unicode_ci`? We should document this separately somewhere.

Reply:
It was `Illegal mix of collations ...` We documented that on release notes but do you think we should do that in the code as well? I didn't removed the expected one because it answers the question what is the supported collation.